config/v1/types_cluster_version.go

```diff
@@ -40,13 +40,29 @@ type ClusterVersionSpec struct {
 	// +required
 	ClusterID ClusterID `json:"clusterID"`
 
-	// desiredUpdate is an optional field that indicates the desired value of
-	// the cluster version. Setting this value will trigger an upgrade (if
-	// the current version does not match the desired version). The set of
-	// recommended update values is listed as part of available updates in
-	// status, and setting values outside that range may cause the upgrade
-	// to fail. You may specify the version field without setting image if
-	// an update exists with that version in the availableUpdates or history.
+	// desiredUpdate is an optional field that requests cluster updates
+	// when the requested release diverges from the cluster's current
+	// target.
+	//
+	// Several update properties are involved in selecting a target
+	// release image. The set of recommended update values is listed as
+	// part of available updates in status, and setting values outside
+	// that range may cause the upgrade to fail. In some cases where
+	// the set of available updates does not include the desired target
+	// release image, setting the 'image' property extends the set of
+	// available updates to include the provided image. The
+	// cluster-version operator then filters the possibly-extended set
+	// of available updates to find an entry that matches all of the
+	// provided desired properties. Currently 'image', 'version', and
```

Comment:
This isn't what actually happens and we discussed the prereqs needed before we make this change.

Right now, the goal of this struct is produce an image.  The cases are.

1. image is specified, version is specified, architecture is specified.  API validation error.
2. image is specified, version is specified, architecture is not specified.   You should not do this.  version is silently ignored and image is used.
3. image is specified, version is not specified, architecture is specified.  API validation error
4. image is specified, version is not specified, architecture is not specified.  image is used.
5. image is not specified, version is specified, architecture is specified.  version and desired architecture are used to select an image.
6. image is not specified, version is specified, architecture is not specified.  version and current architecture are used to select an image.
7. image is not specified, version is not specified, architecture is specified.  API validation error
8. image is not specified, version is not specified, architecture is not specified.  API validation error